File: socialByJava/src/main/java/com/example/socialbyjava/etc/GlobalExceptionHandler.java
package com.example.socialbyjava.etc;

import com.example.socialbyjava.etc.enums.TokenErrorResult;
import com.example.socialbyjava.etc.enums.UserErrorResult;
import com.example.socialbyjava.etc.error.TokenException;
import com.example.socialbyjava.etc.error.UserException;
import com.example.socialbyjava.etc.response.ApiResponse;
import com.example.socialbyjava.etc.response.BaseErrorCode;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;
import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;

@RestControllerAdvice
@Slf4j
public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {

    @ExceptionHandler(TokenException.class)
    public ResponseEntity<ApiResponse<BaseErrorCode>> handleTokenException(TokenException e) {
        TokenErrorResult errorResult = e.getTokenErrorResult();
        return ApiResponse.onFailure(errorResult);
    }

    @ExceptionHandler(UserException.class)
    public ResponseEntity<ApiResponse<BaseErrorCode>> handleUserException(UserException e) {
        UserErrorResult errorResult = e.getUserErrorResult();
        return ApiResponse.onFailure(errorResult);
    }
}

File: Spring/src/main/kotlin/com/example/spring/GlobalExceptionHandler.kt
package com.example.spring

import com.example.spring.domain.base.error.BaseErrorCode
import com.example.spring.domain.base.response.web.ApiResponse
import com.example.spring.domain.token.error.exception.TokenException
import com.example.spring.domain.user.error.exception.UserErrorResult
import com.example.spring.domain.user.error.exception.UserException
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice


@RestControllerAdvice
class GlobalExceptionHandler {

    @ExceptionHandler(TokenException::class)
    fun handleTokenException(e: TokenException): ResponseEntity<ApiResponse<BaseErrorCode?>> {
        val errorResult = e.tokenErrorResult
        return ApiResponse.onFailure(errorResult!!)
    }

    @ExceptionHandler(UserException::class)
    fun handleUserException(e: UserException): ResponseEntity<ApiResponse<BaseErrorCode?>> {
        val errorResult: UserErrorResult = e.userErrorResult
        return ApiResponse.onFailure(errorResult)
    }
}